extensions/api/v1alpha1/sandboxtemplate_types.go

```diff
@@ -31,13 +31,62 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	sandboxv1alpha1 "sigs.k8s.io/agent-sandbox/api/v1alpha1"
 )
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 // Important: Run "make" to regenerate code after modifying this file
 
+// NetworkPolicySpec defines the desired state of the NetworkPolicy.
+type NetworkPolicySpec struct {
+	Enabled                    bool             `json:"enabled,omitempty"`
+	IngressControllerSelectors *IngressSelector `json:"ingressControllerSelectors,omitempty"`
+	IngressFromIPBlocks        []IPBlock        `json:"ingressFromIPBlocks,omitempty"`
+	AdditionalIngressRules     []IngressRule    `json:"additionalIngressRules,omitempty"`
+	AdditionalEgressRules      []EgressRule     `json:"additionalEgressRules,omitempty"`
+}
+
+// IngressSelector defines selectors for an in-cluster ingress controller.
+type IngressSelector struct {
+	// NamespaceSelector matches the namespace of the ingress controller.
+	// Uses standard LabelSelector (supports matchLabels and matchExpressions).
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
What's the default behavior if NamespaceSelector isn't specified?

Reply:
If NamespaceSelector is nil but PodSelector is provided, the standard Kubernetes NetworkPolicy behavior applies: it selects Pods in the SAME namespace as the Sandbox.